testkit/src/main/scala/org/bitcoins/rpc/BitcoindRpcTestUtil.scala

```diff
@@ -213,62 +243,193 @@ trait BitcoindRpcTestUtil extends BitcoinSLogger {
+    (client1, client2)
+  }
 
-      if (try3.isFailure) {
-        logger.error(
-          s"Failed to connect client1 and client 2 ${try3.failed.get.getMessage}")
-        deleteNodePair(client1, client2)
-        throw try3.failed.get
+  /**
+    * Returns a pair of [[org.bitcoins.rpc.client.common.BitcoindRpcClient BitcoindRpcClient]]
+    * that are connected with 100 blocks in the chain
+    */
+  def createNodePair(
+      port1: Int = randomPort,
+      rpcPort1: Int = randomPort,
+      port2: Int = randomPort,
+      rpcPort2: Int = randomPort)(implicit system: ActorSystem): Future[
+    (BitcoindRpcClient, BitcoindRpcClient)] = {
+    implicit val executionContext: ExecutionContext = system.dispatcher
+    val (client1, client2) = createUnconnectedNodePair()
+    client1
+      .addNode(client2.getDaemon.uri, AddNodeArgument.Add)
+      .map(_ => {
+        val try3 = Try(awaitConnection(client1, client2))
+        if (try3.isFailure) {
+          deleteNodePair(client1, client2)
+          throw try3.failed.get
+        }
+      })
+      .map { _ =>
+        client1.generate(100)
+      }
+      .map { _ =>
+        {
+          val try4 = Try(awaitSynced(client1, client2))
+          if (try4.isFailure) {
+            deleteNodePair(client1, client2)
+            throw try4.failed.get
+          }
+          (client1, client2)
+        }
       }
-      client1.generate(100).map { _ =>
-        val try4 = Try(awaitSynced(client1, client2))
+  }
 
-        if (try4.isFailure) {
-          deleteNodePair(client1, client2)
-          throw try4.failed.get
+  def createRawCoinbaseTransaction(
+      sender: BitcoindRpcClient,
+      receiver: BitcoindRpcClient,
+      amount: Bitcoins = Bitcoins(1))(
+      implicit executionContext: ExecutionContext): Future[Transaction] = {
+    sender.generate(2).flatMap { blocks =>
+      sender.getBlock(blocks(0)).flatMap { block0 =>
+        sender.getBlock(blocks(1)).flatMap { block1 =>
+          sender.getTransaction(block0.tx(0)).flatMap { transaction0 =>
+            sender.getTransaction(block1.tx(0)).flatMap { transaction1 =>
+              val input0 =
+                TransactionOutPoint(transaction0.txid.flip,
+                                    UInt32(transaction0.blockindex.get))
+              val input1 =
+                TransactionOutPoint(transaction1.txid.flip,
+                                    UInt32(transaction1.blockindex.get))
+              val sig: ScriptSignature = ScriptSignature.empty
+              receiver.getNewAddress.flatMap { address =>
+                sender.createRawTransaction(
+                  Vector(TransactionInput(input0, sig, UInt32(1)),
+                         TransactionInput(input1, sig, UInt32(2))),
+                  Map(address -> amount))
+              }
+            }
+          }
         }
-        (client1, client2)
       }
     }
   }
 
+  def signRawTransaction(
```

Comment:
This util method is needed because V16 and V17 has different APIs for signing raw transactions (V17 deprecated the `signrawtransaction` RPC and split it into `signrawtransactionwithwallet` and `signrawtransactionwithkey`)

Quite a few tests sign raw transactions, and I figured it was best to use this util methods in that case, and then implement tests specifically for the V16 and V17 RPC calls

Reply:
Let's add this comment to the source code